packages/roosterjs-editor-api/lib/format/clearFormat.ts

```diff
@@ -1,307 +1,339 @@
-                    }
-                });
-            });
-        }
-    });
-
-    if (!isDefaultFormatEmpty) {
-        if (defaultFormat.fontFamily) {
-            setFontName(editor, defaultFormat.fontFamily);
-        }
-        if (defaultFormat.fontSize) {
-            setFontSize(editor, defaultFormat.fontSize);
-        }
-        if (defaultFormat.textColor) {
-            const setColorIgnoredElements = editor.queryElements<HTMLElement>(
-                'a *, a',
-                QueryScope.OnSelection
-            );
-
-            let shouldApplyInlineStyle =
-                setColorIgnoredElements.length > 0
-                    ? (element: HTMLElement) => setColorIgnoredElements.indexOf(element) == -1
-                    : null;
-
-            if (defaultFormat.textColors) {
-                setTextColor(editor, defaultFormat.textColors, shouldApplyInlineStyle);
-            } else {
-                setTextColor(editor, defaultFormat.textColor, shouldApplyInlineStyle);
-            }
-        }
-        if (defaultFormat.backgroundColor) {
-            if (defaultFormat.backgroundColors) {
-                setBackgroundColor(editor, defaultFormat.backgroundColors);
-            } else {
-                setBackgroundColor(editor, defaultFormat.backgroundColor);
-            }
-        }
-        if (defaultFormat.bold) {
-            toggleBold(editor);
-        }
-        if (defaultFormat.italic) {
-            toggleItalic(editor);
-        }
-        if (defaultFormat.underline) {
-            toggleUnderline(editor);
-        }
-    }
-}
-
-function removeStyles(tag: string, node: HTMLElement, isDefaultFormatEmpty: boolean) {
-    if (TAGS_TO_STOP_UNWRAP.indexOf(tag) == -1) {
-        STYLES_TO_REMOVE.forEach(style => node.style.removeProperty(style));
-
-        // when default format is empty, keep the HTML minimum by removing style attribute if there's no style
-        // (note: because default format is empty, we're not adding style back in)
-        if (isDefaultFormatEmpty && node.getAttribute('style') === '') {
-            node.removeAttribute('style');
-        }
-    }
-}
-
-/**
- * Clear the format in current selection, after cleaning, the format will be
- * changed to default format. The format that get cleaned include B/I/U/font name/
- * font size/text color/background color/align left/align right/align center/superscript/subscript
- * @param editor The editor instance
- * @param formatType type of format to apply
- */
-export default function clearFormat(
-    editor: IEditor,
-    formatType: ClearFormatMode | CompatibleClearFormatMode = ClearFormatMode.Inline
-) {
-    switch (formatType) {
-        case ClearFormatMode.Inline:
-            clearInlineFormat(editor);
-            break;
-        case ClearFormatMode.Block:
-            clearBlockFormat(editor);
-            break;
-        default:
-            clearAutoDetectFormat(editor);
-    }
-}
+import blockFormat from '../utils/blockFormat';
```

Comment:
try not change the line end so we can do better review

Reply:
welp, https://github.com/microsoft/roosterjs/pull/1415